### PR TITLE
Arabic feature based on #1409

### PR DESCRIPTION
**Pull request by kisvegabor**
I've created a new Pull request to better see the changes.

### CODE REVIEW

File: src/lv_widgets/lv_cont.c [hunk 1]
```diff
@@ -649,7 +649,6 @@ static void lv_cont_layout_pretty(lv_obj_t * cont)
  */
 static void lv_cont_layout_grid(lv_obj_t * cont)
 {
-
```
Comment (embeddedt) [inline src/lv_widgets/lv_cont.c:652]: Is this a leftover from past merge conflicts?

**Comment by embeddedt** [comment]
@hamidrm Can `lv_font_ap_18` be added to the [built_in_font](https://github.com/littlevgl/lvgl/tree/dev-7.0/scripts/built_in_font) directory like the other fonts?

File: src/lv_widgets/lv_cont.c (hunk 1 above)
Comment (kisvegabor) [reply src/lv_widgets/lv_cont.c:652]: Do you mean the removed empty line?
If so, honestly, I don't know. :) Maybe I just saw the empty line while integrating this feature and removed it.

**Comment by kisvegabor** [comment]
> Can lv_font_ap_18 be added to the built_in_font directory like the other fonts?

I see a few things to discuss about it:
- I think it's enough to have an Arabic font in one size as an example. I vote for 16 px instead. (18 px might be too large on low DPI screens)
- Which characters shall it include? Hebrew, Arabic, Persian?
- We should give a more descriptive name, such as `lv_font_arabic_16`.

**Comment by kisvegabor** [comment]
I'd like to merge this PR but it was great to really add a built-in font. Can you offer a typical and popular Arabic/Persian/Hebrew font?

**Comment by amirgon** [comment]
@kisvegabor - Do you just want a popular font, or one which is also legal to convert and distribute?  

There are lots of fonts out there (from Microsoft for example), but I'm not sure it's legal to distribute them with lvgl. I don't know much about this, but I think that legally you can **use** many fonts, but not **distribute** them.

**Comment by ali-rostami** [comment]
> I'd like to merge this PR but it was great to really add a built-in font. Can you offer a typical and popular Arabic/Persian/Hebrew font?

I'm not sure what's your metrics to choose a one, but **[here](https://github.com/rastikerdar/parastoo-font)** there is a nice-at least in my opinion-, free and open-source Persian font.
PS: In between Persian and Arabic font, It's better to use a Persian one. because the Persian alphabet has 4 more characters. so we can have both of them by this.

**Comment by kisvegabor** [comment]
@ali-rostami This font looks great - as far as I can judge. :slightly_smiling_face: 
@amirgon Do you agree to go with it?

Finally which ranges shall we include in the font to cover the most common Arabic, Persian and Hebrew characters?

**Comment by ali-rostami** [comment]
> @ali-rostami This font looks great - as far as I can judge. 🙂
> 
> Finally which ranges shall we include in the font to cover the most common Arabic, Persian and Hebrew characters?

Hebrew alphabet is totally different from the Arabic and Persian alphabet. The similarity is just all of them are right to left languages.
And as I said, Arabic and Persian have a similar alphabet except that the Persian alphabet has just 4 more characters. 
So If you use the Persian font -the link I provided for you- we can have both Arabic and Persian alphabet without any problem.

**Comment by kisvegabor** [comment]
@ali-rostami Got it, but I can feed a rang or a list of symbols to the font converter. So my question is: is the [0x0600..0x6FF](https://en.wikipedia.org/wiki/Arabic_script_in_Unicode) range enough, or too much?

**Comment by ali-rostami** [comment]
> @ali-rostami Got it, but I can feed a rang or a list of symbols to the font converter. So my question is: is the [0x0600..0x6FF](https://en.wikipedia.org/wiki/Arabic_script_in_Unicode) range enough, or too much?

Oops, Interesting thing I didn't know. Yeah, the range is good. If you use 0x600 - 0x6FF we can have Persian, Arabic, and even Afghani signs too.

**Comment by kisvegabor** [comment]
@ali-rostami Thank you!

I've just checked DejaVu which includes both Hebrew and Persian characters:
image

What would you think about using it?

**Comment by ali-rostami** [comment]
> What would you think about using it?

It's OK. thanks

**Comment by hamidrm** [comment]
@kisvegabor This range [0600–06FF] contains just 'Arabic General Unicode Symbols' (Same as isolated letters form). In addition to the [0600–06FF] range, Your font must be included "Arabic Presentation Forms-A (FB50–FDFF, 611 characters)" and "Arabic Presentation Forms-B (FE70–FEFF, 141 characters)".
As mentioned [here ](https://github.com/littlevgl/lvgl/pull/1475/commits/95cad47068c174cd2600e03b6b3138f7e16aa64b):
```
bool lv_bidi_letter_is_rtl(uint32_t letter){
    if(letter >= 0x5d0 && letter <= 0x5ea) return true;
    if(letter == 0x202E) return true;               /*Unicode of LV_BIDI_RLO*/
    /* Check for Persian and Arabic characters [https://en.wikipedia.org/wiki/Arabic_script_in_Unicode]*/
    if(letter >= 0x600 && letter <= 0x6FF) return true;
    if(letter >= 0xFB50 && letter <= 0xFDFF) return true;
    if(letter >= 0xFE70 && letter <= 0xFEFF) return true;
    return false;
}
```
@embeddedt Sorry for my late reply. I'm not sure it's legal to add it (due to using the Microsoft font).

**Comment by amirgon** [comment]
> I've just checked DejaVu which includes both Hebrew and Persian characters

@kisvegabor DejaVu is fine as Hebrew font.

**Comment by ali-rostami** [comment]
> This range [0600–06FF] contains just 'Arabic General Unicode Symbols' (Same as isolated letters form). In addition to the [0600–06FF] range, Your font must be included "Arabic Presentation Forms-A (FB50–FDFF, 611 characters)" and "Arabic Presentation Forms-B (FE70–FEFF, 141 characters)".

Hmm, totally out of my information. 
@hamidrm I wonder why it is like this when we have all of the characters in the range of  [0600–06FF].

I checked the DejaVu, It has "Arabic Presentation Forms-A (FB50–FDFF, 611 characters)" and "Arabic Presentation Forms-B (FE70–FEFF, 141 characters)" too. so I think it's OK to use it. what do you think @hamidrm ?

**Comment by kisvegabor** [comment]
I've just added a built-in DejaVu font for Hebrew letters and Persian letters as suggested by @hamidrm. See [here](https://github.com/littlevgl/lvgl/blob/dev-7.0/src/lv_font/lv_font_dejavu_16_persian_hebrew.c).

Also merged this  PR to `dev-7.0`.

Thank you very much for your contribution!

**Pull request merged**